packages/core/src/engine/Autodiff.test.ts

```diff
@@ -270,3 +296,89 @@ const gradGraph0 = (): ad.Graph => {
 
   return graph;
 };
+
+//#endregion
+
+describe("polyRoots tests", () => {
+  test("degree 1", () => {
+    const [z] = polyRoots([input({ key: 0, val: 0 })]);
+    const g = primaryGraph(z);
+    const f = genCode(g);
+    const x = 42;
+    expect(f([x])).toEqual({ gradient: [-1], primary: -x, secondary: [] });
+  });
+
+  type F = (v: VarAD, w: VarAD) => VarAD;
+
+  const testQuadratic = (f1: F, f2: F) => {
```

Comment:
Super cool test case. Took me a while to figure out this is testing if the polyroot solution is the same as the closed form formula for both roots of quadratic equations. Would be great to add a short comment that explains what this function is doing.

My understanding is that the polyroots comp graph (`implicit`) finds the roots of `ax^2 + bx + c = 0`, where `a = PI^2` and `b = - PI - E`?

Reply:
Also, is that a monic polynomial? Or the monic part is just what the system does internally?

Reply:
> Took me a while to figure out this is testing if the polyroot solution is the same as the closed form formula for both roots of quadratic equations. Would be great to add a short comment that explains what this function is doing.

Good point! I'll add a comment about that.

> My understanding is that the polyroots comp graph (`implicit`) finds the roots of `ax^2 + bx + c = 0`, where `a = PI^2` and `b = - PI - E`?
>
> Also, is that a monic polynomial? Or the monic part is just what the system does internally?

Not quite; the `polyRoots` comp graph is finding the roots of c + bx + x², where c = πe and b = -(π + e).

Reply:
aha interesting! I was reading it backwards apparently!

Reply:
I was lazy enough to think the `4ac` term in the closed form solution explained the ordering. Well multiplication commutes!